mlflow/store/model_registry/rest_store.py

```diff
@@ -399,3 +399,6 @@ def get_model_version_by_alias(self, name, alias):
         req_body = message_to_json(GetModelVersionByAlias(name=name, alias=alias))
         response_proto = self._call_endpoint(GetModelVersionByAlias, req_body)
         return ModelVersion.from_proto(response_proto.model_version)
+
+    def _await_model_version_creation(self, mv, await_creation_for):
```

Comment:
Note: with this PR, only rest stores will wait for created model versions to be ready. This is because, for non-rest stores like file store and SQL alchemy store, the model versions are instantly ready.

Reply:
@BenWilson2 @harupy let me know if this is a sane approach. I'm not sure if users have extended the `AbstractStore` directly in ways that would need for this awaiting.

Reply:
Actually, seeing the test failures, just to be safe I'm going to move the await functionality to the base abstract class.